### PR TITLE
RenameDriver with TextFirst active

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2129.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs [hunk 1]
```diff
@@ -763,5 +763,24 @@ public void AggregateFieldTypeDisplayName()
             var recordType = RecordType.Empty().Add("numLogical", FormulaType.Number, "numDisplay");
             Assert.Equal("numDisplay", recordType.GetFieldTypes().First().DisplayName);
         }
+
+        [Theory]
+        [InlineData("=var1", true)]
+        [InlineData("Var1", false)]
+        [InlineData("text ${var1} text", true)]
+        [InlineData("={'prop1':var1}", true)]
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs:771]: >={'prop1':var1} [](http://example.com/codeflow?start=21&length=15)

can we assert the final string for an exact match? 

(ie, ensure things like we've preserver the TextFirst status and didn't convert to non-text first)

File: src/libraries/Microsoft.PowerFx.Core/Localization/ExpressionLocalizationHelper.cs [hunk 2]
```diff
@@ -43,7 +50,12 @@ internal static string ConvertExpression(string expressionText, RecordType param
             }
 
             return Span.ReplaceSpans(expressionText, worklist);
-        }
+        }
+
+        internal static string ConvertExpression(string expressionText, RecordType parameters, BindingConfig bindingConfig, INameResolver resolver, IBinderGlue binderGlue, CultureInfo culture, Features flags, bool toDisplay)
+        {
+            return ConvertExpression(expressionText, parameters, bindingConfig, resolver, binderGlue, new ParserOptions() { Culture = culture }, flags, toDisplay);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Localization/ExpressionLocalizationHelper.cs:57]: > new ParserOptions() { Culture = culture } [](http://example.com/codeflow?start=101&length=42)

nice! This is a good example of chaining.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**